index.bs

```diff
@@ -3680,31 +3682,29 @@ To <dfn>trigger event-level attribution</dfn> given an [=attribution trigger=] |
     :: null
     : [=attribution rate-limit record/event-level report ID=]
     :: |report|'s [=event-level report/report ID=]
-1. If the result of running [=check if attribution should be blocked by rate limits=]
-    with |trigger|, |sourceToAttribute|, and |rateLimitRecord| is not null, return it.
-1. If |sourceToAttribute|'s [=attribution source/event-level attributable=] value
-    is false:
-     1. Return the [=triggering result=] ("<code>[=triggering status/dropped=]</code>",
-         ("<code>[=trigger debug data type/trigger-event-excessive-reports=]</code>", |report|)).
-1. If the result of running [=maybe replace event-level report=] with |sourceToAttribute| and |report| is:
-    <dl class="switch">
-    : "<code>[=event-level-report-replacement result/add-new-report=]</code>"
-    ::
-         1. Let |numMatchingReports| be the number of entries in the [=event-level report cache=] whose
-             [=event-level report/attribution destinations=] [=set/contains=] |trigger|'s [=attribution trigger/attribution destination=].
-         1. If |numMatchingReports| is greater than or equal to the user agent's [=max event-level reports per attribution destination=]:
+1. If |sourceToAttribute|'s [=attribution source/number of event-level reports=] is less than
```

Comment:
Can we still have these checks in "maybe replace event-level report", and perform the attribution rate-limit check after that?  I understand that in the case of replacement, it's guaranteed to be pass, but it seems to be clearer. Otherwise, we probably need a note why we don't check attribution rate-limits in that case.

Reply:
I thought about it, but it seems to complicate the return result and parameters passed to "maybe replace event-level report" and doesn't seem directly related to that procedure. I'm happy to add a note because the assumption about the relationship between max number of event-level reports and attribution rate limit would still seem to be there and perhaps should be explicit.

Reply:
Actually, attribution rate limit is vendor specific and they could choose to make it lower than max reports, in which case this change wouldn't work, right?

Reply:
I'm confusing myself thinking about the scenarios for the limits :). @linnan-github could you please explain why you think a note is necessary? Is there a scenario where this logic breaks?

Reply:
I’m a little confused, what I meant to say is just move attribution rate-limit check later and only  check this when the result is add new report from “maybe replace the event-level report”. This aligns with how we deal with the storage limit. The old rate-limit record would have been deleted in case of replacement and the check should pass.

I think at least adding a note to explain why we don’t need this check when the number of event-level reports equals the limit.

Reply:
@linnan-github what would happen under your proposal if, as a vendor, I set attribution rate limit to 10 (assuming the constant, max event reports, stays at 20)?

Reply:
Ah, that seems a simpler change than what I've done here :) Thanks!

Reply:
Pushed. Is this what you had in mind?